src/amo/components/Header/index.js

```diff
@@ -0,0 +1,93 @@
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+
+import Link from 'amo/components/Link';
+import SearchForm from 'amo/components/SearchForm';
+import AuthenticateButton from 'core/components/AuthenticateButton';
+import translate from 'core/i18n/translate';
+import Button from 'ui/components/Button';
+import Icon from 'ui/components/Icon';
+
+import 'mozilla-tabzilla/css/_tabzilla.scss';
+import './styles.scss';
+
+
+export class HeaderBase extends React.Component {
+  static propTypes = {
+    i18n: PropTypes.object.isRequired,
+    isHomePage: PropTypes.bool.isRequired,
+    lang: PropTypes.string.isRequired,
+    location: PropTypes.object.isRequired,
+    SearchFormComponent: PropTypes.node.isRequired,
+    query: PropTypes.string,
+  }
+
+  static defaultPropTypes = {
+    SearchFormComponent: SearchForm,
+    isHomePage: false,
+  }
+
+  render() {
+    const {
+      SearchFormComponent,
+      i18n,
+      isHomePage,
+      lang,
+      location,
+      query,
+    } = this.props;
+    const headerLink = (
+      <Link className="Header-title" to="/">
+        <Icon className="Header-addons-icon" name="extensions-complex" />
+        {i18n.gettext('Firefox Add-ons')}
+      </Link>
+    );
+
+    return (
+      <div className="Header">
+        <div className="Header-top-row">
+          <div className="Header-buttons">
+            {/*
+              We don't use the Link component here because we only want to
```

Comment:
Yeah that's way better and means not having to map `state.api` to the `props` of the component using the `<Link>` too. Will do! 👍